### PR TITLE
HPCC-15355 Fix various windows compile errors and warnings

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@jakesmith please review.

**Comment by richardkchapman** [comment]
@jakesmith Please review

File: thorlcr/activities/thactivityutil.cpp [hunk 1]
```diff
@@ -257,9 +257,6 @@ class CRowStreamLookAhead : public CSimpleInterfaceOf<IStartableEngineRowStream>
         }
     }
 };
-#ifdef _MSC_VER
-#pragma warning(pop)
-#endif
```
Comment (jakesmith) [inline thorlcr/activities/thactivityutil.cpp]: Was this just a ancient legacy code leftover?

File: thorlcr/thorcodectx/thcodectx.hpp [hunk 2]
```diff
@@ -50,6 +50,7 @@ class thcodectx_decl CThorCodeContextBase : public CSimpleInterfaceOf<ICodeConte
 
 public:
     CThorCodeContextBase(CJobChannel &jobChannel, ILoadedDllEntry &_querySo, IUserDescriptor &_userDesc);
+    IMPLEMENT_IINTERFACE_USING(CSimpleInterfaceOf<ICodeContextExt>) // This is strangely required by visual studio to ensure Release() is resolved
```
Comment (jakesmith) [inline thorlcr/thorcodectx/thcodectx.hpp]: Odd.. aren't there are other cases?

**Comment by jakesmith** [comment]
@ghalliday - some incidental questions, but good to merge.

**Pull request merged**